ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1138,24 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    /* Session version and negotiated protocol version should match
+     * (This may not be the case when reusing old session)
+     */
+    if (s->version != s->session->ssl_version) {
```

Comment:
So now I don't understand the comment! :-) If we attempted to reuse an old session, but failed, then `s->hit == 0` and `s->version == s->session->ssl_version`? So how can this ever fail except through server error?